### PR TITLE
Transparent window and easy installation issues fixed

### PR DESCRIPTION
**Pull request by ghost**
The tool is now transparent and locked to the top. While installation just requires the build folder which the exe is inside.

### CODE REVIEW

File: mainWindow.py [hunk 1]
```diff
@@ -25,23 +25,25 @@ def setupUi(self, MainWindow):
...
         self.label.setObjectName("label")
+        self.label.setWordWrap(True)
+        self.label.setStyleSheet("QLabel { color: white; qproperty-alignment: AlignCenter; background-color: rgba(0, 0, 0, 0.7); font-size: 15px; font-weight: bold; }")
         MainWindow.setCentralWidget(self.centralwidget)
 
-        #self.setAttribute(Qt.WA_TranslucentBackground)
-        #self.setStyleSheet("background:transparent")
+        self.setWindowFlags(Qt.FramelessWindowHint | Qt.WindowStaysOnTopHint)
+        self.setAttribute(Qt.WA_TranslucentBackground)
         #self.setAttribute(Qt.WA_NoSystemBackground)
```
Comment (gmelodie) [inline mainWindow.py:45]: You can go ahead and remove this comment

File: mainWindow.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Ui_MainWindow(object):
     def setupUi(self, MainWindow):
         MainWindow.setObjectName("MainWindow")
         MainWindow.setWindowModality(QtCore.Qt.NonModal)
-        MainWindow.resize(202, 200)
+        MainWindow.resize(400, 400)
         MainWindow.setStyleSheet("")
         MainWindow.setProperty("WindowStaysOnTopHint", True)
         self.centralwidget = QtWidgets.QWidget(MainWindow)
```
Comment (gmelodie) [inline mainWindow.py:19]: No changes should be commited in this file as it's generated automatically from the `.ui` file (which is used by QtDesigner). Make all the changes of style on the `hbc.py` file

File: mainWindow.py [hunk 3]
```diff
@@ -25,23 +25,25 @@ def setupUi(self, MainWindow):
         self.heroComboBox.setEditable(False)
         self.heroComboBox.setObjectName("heroComboBox")
         self.heroComboBox.addItem("")
+        self.heroComboBox.setStyleSheet("QComboBox { color: white; background-color: rgba(0, 0, 0, 0.7); font-size: 15px; }")
```
Comment (gmelodie) [inline mainWindow.py:28]: Same as above

File: mainWindow.py [hunk 4]
```diff
@@ -25,23 +25,25 @@ def setupUi(self, MainWindow):
...
         self.talentComboBox = QtWidgets.QComboBox(self.centralwidget)
         self.talentComboBox.setGeometry(QtCore.QRect(0, 30, 201, 25))
         self.talentComboBox.setMinimumContentsLength(0)
         self.talentComboBox.setObjectName("talentComboBox")
         self.talentComboBox.addItem("")
+        self.talentComboBox.setStyleSheet("QComboBox { color: white; background-color: rgba(0, 0, 0, 0.7); font-size: 15px; }")
```
Comment (gmelodie) [inline mainWindow.py:34]: Same as above

File: mainWindow.py [hunk 5]
```diff
@@ -25,23 +25,25 @@ def setupUi(self, MainWindow):
...
         self.label = QtWidgets.QLabel(self.centralwidget)
-        self.label.setGeometry(QtCore.QRect(40, 70, 141, 141))
+        self.label.setGeometry(QtCore.QRect(0, 70, 201, 181))
```
Comment (gmelodie) [inline mainWindow.py:36]: Same as above

File: mainWindow.py [hunk 6]
```diff
@@ -25,23 +25,25 @@ def setupUi(self, MainWindow):
...
         self.label.setText("")
         self.label.setObjectName("label")
+        self.label.setWordWrap(True)
```
Comment (gmelodie) [inline mainWindow.py:39]: Same as above

File: mainWindow.py [hunk 7]
```diff
@@ -25,23 +25,25 @@ def setupUi(self, MainWindow):
...
+        self.label.setStyleSheet("QLabel { color: white; qproperty-alignment: AlignCenter; background-color: rgba(0, 0, 0, 0.7); font-size: 15px; font-weight: bold; }")
         MainWindow.setCentralWidget(self.centralwidget)
 
-        #self.setAttribute(Qt.WA_TranslucentBackground)
-        #self.setStyleSheet("background:transparent")
+        self.setWindowFlags(Qt.FramelessWindowHint | Qt.WindowStaysOnTopHint)
```
Comment (gmelodie) [inline mainWindow.py:43]: Same as above

File: hbc.ui [hunk 8]
```diff
@@ -81,7 +81,7 @@
      </rect>
     </property>
     <property name="text">
-     <string/>
+     <string></string>
```
Comment (gmelodie) [inline hbc.ui:84]: Why is this changed? I believe it works both ways, doesn't it?

File: hbc.py [hunk 9]
```diff
@@ -1,6 +1,9 @@
 import sys
 from PyQt5.QtWidgets import *
 import mainWindow
+from PyQt5 import QtCore, QtWidgets, QtGui
+from PyQt5.QtGui import QIcon,QFont,QPixmap,QPalette
```
Comment (gmelodie) [inline hbc.py:5]: Why are you importing all of this if you only used QPoint? Sorry if it's a lame question, I'm new with PyQt

File: mainWindow.py [hunk 10]
```diff
@@ -13,7 +13,7 @@ class Ui_MainWindow(object):
     def setupUi(self, MainWindow):
         MainWindow.setObjectName("MainWindow")
         MainWindow.setWindowModality(QtCore.Qt.NonModal)
-        MainWindow.resize(202, 200)
+        MainWindow.resize(400, 400)
```
Comment (gmelodie) [inline mainWindow.py:16]: Whis has the size changed?

**Review by gmelodie — CHANGES_REQUESTED** [review]
Window now draggable (fdc98c5a319751e97415893fb1481128648dd2fc): as of the last time I tested window was already draggable, I'm not sure if this is a feature that needs to be implemented

- Please remove the build directory from versioning (**do not upload build files**).

- After removing the build directory I think it would be nice to add it to .gitignore in order to avoid that in the future

Overall, thanks for the contributions! This helps a lot :smile:

File: mainWindow.py (hunk 2 above)
Comment (gmelodie) [reply mainWindow.py:19]: You can put this under the comment `# Custom editions on QMainWindow` on `hbc.py`, just like I did :)

**Comment by ghost** [comment]
At first when the window was transparent, it couldn't be moved. So that's why I had to add the draggable feature again. And thanks for the useful comments! I'll remember them for future pull requests.

**Comment by gmelodie** [comment]
Aaaaah makes sense. Awesome help, really :smile:
